File: backend/src/main/java/com/backend/ord/config/properties/OpenAIProperties.java
package com.backend.ord.config.properties;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix = "openai")
public class OpenAIProperties {
    private String apiKey;
    private String apiUrl;
    private String gptModel;
    private int maxTokens;
    private float temperature;

    public String getAuthenticationHeaderValue() {
        return "Bearer " + apiKey;
    }

    public String getApiKey() {
        return this.apiKey;
    }

    public String getApiUrl() {
        return this.apiUrl;
    }

    public String getGptModel() {
        return this.gptModel;
    }

    public int getMaxTokens() {
        return this.maxTokens;
    }

    public float getTemperature() {
        return this.temperature;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }

    public void setApiUrl(String apiUrl) {
        this.apiUrl = apiUrl;
    }

    public void setGptModel(String gptModel) {
        this.gptModel = gptModel;
    }

    public void setMaxTokens(int maxTokens) {
        this.maxTokens = maxTokens;
    }

    public void setTemperature(float temperature) {
        this.temperature = temperature;
    }
}

File: backend/src/main/kotlin/com/backend/ord/config/properties/OpenAIProperties.kt
package com.backend.ord.config.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@ConfigurationProperties(prefix = "openai")
open class OpenAIProperties(
    var apiKey: String = "",
    var apiUrl: String = "",
    var gptModel: String = "",
    var maxTokens: Int = 0,
    var temperature: Float = 0f
) {
    val authenticationHeaderValue: String
        get() = "Bearer $apiKey"
}
